packages/manager/src/features/CloudPulse/Utils/CloudPulseWidgetUtils.ts

```diff
@@ -194,33 +195,52 @@ export const generateGraphData = (props: graphDataOptionsProps) => {
           serviceType,
           unit,
         };
-
-        const dimension = {
-          backgroundColor: color,
-          borderColor: color,
-          data: seriesDataFormatter(transformedData.values, start, end),
-          fill: widgetChartType === 'area',
-          label: getLabelName(labelOptions),
-        };
+        const labelName = getLabelName(labelOptions);
+        const data = seriesDataFormatter(transformedData.values, start, end);
+        const color = colors[index]?.Primary ?? Alias.Chart.Neutral;
+        areas.push({
+          color,
+          dataKey: labelName,
+        });
+
+        data.forEach((d) => {
```

Comment:
Can we destrcuture 'd' for better readability? May be like below below
`data.forEach(([timestamp, value]) => {
  if (value !== null) {
    dimension[timestamp] = {
      ...dimension[timestamp],
      [labelName]: value,
    };
  }
});`
This makes it clear what each element of d represents and removes the need to create extra variables?